woocommerce/woocommerce-android
# Saving analytics cards configuration

Closes: #10699

### Why
As part of Expanded Analytics Hub milestone 2, we plan to add support for customizing the cards we display on the Analytic Hub screen.

### Description
This PR adds support for saving analytics cards configuration in the device local store. It also update the UI to handle configuration changes and act accordingly.

### Testing instructions
TC1
1. Open the app
2. Tap on See more to enter the Analytics Hub screen
3. Tap on the pencil to enter the Customize Analytics screen
4. Check that the Customize Analytics screen is opened with the default configuration (all cards visible)
5. Check that the save button is disabled if no changes are made
6. Update the current configuration by updating the card selection
7. Check that the save button is enabled when the configuration is different than the initial configuration
8. Tap on save configuration
9. Check that after persisting the changes, the app navigates to the Analytics hub screen
10. Tap on the pencil again
11. Check that the screen displays the previous saved configuration.
12. Check that unselected cards are placed at the bottom of the list.

TC2
1. Open the app
2. Tap on See more to enter the Analytics Hub screen
3. Tap on the pencil to enter the Customize Analytics screen
4. Tap on the close button (X)
5. Check that the app navigates back to the Analytics Hub screen

TC3
1. Open the app
2. Tap on See more to enter the Analytics Hub screen
3. Tap on the pencil to enter the Customize Analytics screen
4. Update the current configuration
5. Tap on the close button (X)
6. Check that the app displays a dialog asking if you want to discard the changes
7. Tap on Keep editing
8. Check that the dialog dismisses
9. Tap on the close button again
10. Check that the discard changes dialog is displayed again
11. Tap on Discard
12. Check that the app navigates back to the Analytics Hub screen
13. Tap on the pencil again and check that the changes from step 4 were not persisted

### Images/gif

https://github.com/woocommerce/woocommerce-android/assets/18119390/3cd3df3d-f040-4686-b946-ef2c412d1578
